integration-tests/smoke/ccip_batching_test.go

```diff
@@ -0,0 +1,295 @@
+	"github.com/smartcontractkit/chainlink/deployment"
+	ccdeploy "github.com/smartcontractkit/chainlink/deployment/ccip"
+	"github.com/smartcontractkit/chainlink/integration-tests/ccip-tests/testsetups"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/multicall3"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/offramp"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/onramp"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/router"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+func Test_CCIPBatching(t *testing.T) {
+	// Setup 3 chains, with 2 lanes going to the dest.
+	lggr := logger.TestLogger(t)
+	ctx := ccdeploy.Context(t)
+	// Will load 3 chains when specified by the overrides.toml or env vars (E2E_TEST_SELECTED_NETWORK).
+	// See e2e-tests.yml.
+	e, _, _ := testsetups.NewLocalDevEnvironmentWithDefaultPrice(t, lggr)
+
+	state, err := ccdeploy.LoadOnchainState(e.Env)
+	require.NoError(t, err)
+
+	allChainSelectors := maps.Keys(e.Env.Chains)
+	require.Len(t, allChainSelectors, 3, "this test expects 3 chains")
+	sourceChain1 := allChainSelectors[0]
+	sourceChain2 := allChainSelectors[1]
+	destChain := allChainSelectors[2]
+	t.Log("All chain selectors:", allChainSelectors,
+		", home chain selector:", e.HomeChainSel,
+		", feed chain selector:", e.FeedChainSel,
+		", source chain selector 1:", sourceChain1,
+		", source chain selector 2:", sourceChain2,
+		", dest chain selector:", destChain,
+	)
+
+	// connect sourceChain1 and sourceChain2 to destChain
+	require.NoError(t, ccdeploy.AddLaneWithDefaultPrices(e.Env, state, sourceChain1, destChain))
+	require.NoError(t, ccdeploy.AddLaneWithDefaultPrices(e.Env, state, sourceChain2, destChain))
+
+	const (
+		numMessages = 5
+	)
+	var (
+		startSeqNum map[uint64]ccipocr3.SeqNum = map[uint64]ccipocr3.SeqNum{
+			sourceChain1: 1,
+			sourceChain2: 1,
+		}
+		endSeqNum map[uint64]ccipocr3.SeqNum = map[uint64]ccipocr3.SeqNum{
+			sourceChain1: ccipocr3.SeqNum(numMessages),
+			sourceChain2: ccipocr3.SeqNum(numMessages),
+		}
+	)
+
+	t.Run("batch data only messages from single source", func(t *testing.T) {
+		err := sendMessages(
+			ctx,
+			t,
+			e.Env.Chains[sourceChain1],
+			e.Env.Chains[sourceChain1].DeployerKey,
+			state.Chains[sourceChain1].OnRamp,
+			state.Chains[sourceChain1].Router,
+			state.Chains[sourceChain1].Multicall3,
+			destChain,
+			numMessages,
+			common.LeftPadBytes(state.Chains[destChain].Receiver.Address().Bytes(), 32),
+		)
+		require.NoError(t, err)
+
+		_, err = ccdeploy.ConfirmCommitWithExpectedSeqNumRange(
+			t,
+			e.Env.Chains[sourceChain1],
+			e.Env.Chains[destChain],
+			state.Chains[destChain].OffRamp,
+			nil,
+			ccipocr3.NewSeqNumRange(startSeqNum[sourceChain1], endSeqNum[sourceChain1]),
+		)
+		require.NoErrorf(t, err, "failed to confirm commit from chain %d", sourceChain1)
```

Comment:
Even though the test is oriented around commit batching, I think it would be worth waiting for messages execution. We don't care about batch sizes for exec, but we want to make sure everything is successfully executed (e.g. this large commit batch is properly processed by the exec)

Reply:
Yeah, I thought about that as well. Exec doesn't have to batch like commit due to gas limits though